iulianIachimov/src/main/java/datacollectiontask/Main.java

```diff
@@ -0,0 +1,123 @@
+        Order orderTwo = new Order(userTwo, ItemListTwo);
+
+        //Create list of orders
+        List<Order> orderList = new ArrayList<>();
+        orderList.add(orderOne);
+        orderList.add(orderTwo);
+
+        //Filter Orders by Type CLOTHES, calling funcion filterOrderByItemType(orderList)
+        List<Order> filteredOrderList = filterOrderByItemType(orderList);
+        System.out.println("the size of filtered order list is " + filteredOrderList.size());
+
+        /*
+        //collection sort, out of tasks
+        System.out.println("userList before sorting");
+        for (User user: userList){
+            System.out.println(user.username);
+        }
+
+        Collections.sort(userList, new Comparator<User>() {
+            @Override
+            public int compare(User userOne, User userTwo) {
+                return userOne.username.compareTo(userTwo.username);
+            }
+        });
+
+        System.out.println("userList after sorting");
+        for (User user: userList){
+            System.out.println(user.username);
+        }
+    */
+
+
+        // Creating the list for users
+        List<User> userList = new ArrayList<>();
+        userList.add(userOne);
+        userList.add(userTwo);
+
+        // Filter users by city
+        List<User> filteredListofUsers = new ArrayList<>();
+        filteredListofUsers = filterUsersByCity(userList, "Chisinau"); // call for viewing the users from Chisinau city
+        System.out.println("The filtered list of users is " + filteredListofUsers.size());
+    }
+
+    // creating function to filter the orders by type CLOTHES
+    public static List<Order> filterOrderByItemType(List<Order> orderList) {
+        List<Order> orderResult = new ArrayList<>(); // creating a list to store the result
+        for (Order order : orderList) {
+            for (Item item : order.items) {
+                if (Type.CLOTHES == item.type) {
+                    orderResult.add(order);
+                    System.out.println("Order " + order.id + " from " + order.date + " contains type Clothes");
+                    //break;
+                }
+            }
+        }
+        return orderResult;
+    }
+
+    // creating function to filter list of users whose city is Chisinau
+    public static List<User> filterUsersByCity(List<User> users, String city) {
+        List<User> userListResult = new ArrayList<>();
+
+        for (User user : users) {
+            for (Map.Entry<String, Address> entry : user.addresses.entrySet()) {
+                if (entry.getValue().city.equals(city)) {
+                    userListResult.add(user);
+                    System.out.println("Username: " + user.username + ", Type of address = " + entry.getKey() + ", city = " + entry.getValue().city);
+                    break;
+                }
+                //   System.out.println("Key = " + entry.getKey() + ", Value = " + entry.getValue().country);
```

Comment:
Make sure that the commented staff can be useful in order to understand the code. Don't leave implementation details under comment.

Reply:
Also pay attention to spaces between code. refer to /n